### ISSUE TITLE
Downloader.init() : Cannot read property 'init' of undefined

### ISSUE THREAD

**Issue by kefahB** [triniwiz/nativescript-downloader#3]
### Which platform(s) does your issue occur on?
Android 
- Android versions : 3.4.1
- Device Nexus_5X_API_23

### Please, provide the following version numbers that your issue occurs with:

- CLI: tns --version : 3.4.1
```
{
	"description": "NativeScript Application",
	"license": "SEE LICENSE IN <your-license-filename>",
	"readme": "NativeScript Application",
	"repository": "<fill-your-repository-here>",
	"nativescript": {
		"id": "org.nativescript.ElHakawati",
		"tns-android": {
			"version": "3.4.1"
		},
		"tns-ios": {
			"version": "3.4.1"
		}
	},
	"dependencies": {
		"1.3.0": "^1.3.0",
		"moment": "^2.20.1",
		"nativescript-audio": "^4.3.3",
		"nativescript-background-http": "^3.1.0",
		"nativescript-downloader": "^1.0.3",
		"nativescript-fontawesome": "^1.0.0",
		"nativescript-snackbar": "^2.0.0",
		"nativescript-sqlite": "^2.0.1",
		"nativescript-theme-core": "~1.0.4",
		"nativescript-zip": "^1.3.5",
		"tns-core-modules": "~3.4.0"
	},
	"devDependencies": {
		"babel-traverse": "6.26.0",
		"babel-types": "6.26.0",
		"babylon": "6.18.0",
		"lazy": "1.0.11",
		"nativescript-dev-typescript": "^0.6.0",
		"typescript": "^2.6.2"
	}
}
```

### Please, tell us how to recreate the issue in as much detail as possible. 
Describe the steps to reproduce it.

### Is there any code involved? 
app.ts
```TS
import * as application from 'tns-core-modules/application';
import { Downloader } from 'nativescript-downloader';
Downloader.init();
application.start({ moduleName: 'main-page' });
```

Hello
I put a nativescript update and suddenly android does not work anymore and I have the following error:
```
System.err: java.lang.RuntimeException: Unable to create application com.tns.NativeScriptApplication: com.tns.NativeScriptException: 
System.err: Error calling module function 
System.err: TypeError: Cannot read property 'init' of undefined
System.err: File: "file:///data/data/org.nativescript.ElHakawati/files/app/tns_modules/nativescript-downloader/downloader.js, line: 17, column: 41
System.err: StackTrace: 
System.err: 	Frame: function:'Downloader.init', file:'file:///data/data/org.nativescript.ElHakawati/files/app/tns_modules/nativescript-downloader/downloader.js', line: 17, column: 42
System.err: 	Frame: function:'', file:'file:///data/data/org.nativescript.ElHakawati/files/app/app.js', line: 45, column: 38
System.err: 	Frame: function:'require', file:'', line: 1, column: 266
System.err: TypeError: Cannot read property 'init' of undefined
System.err: File: "<unknown>, line: 1, column: 265
System.err: StackTrace: 
System.err: 	Frame: function:'Downloader.init', file:'file:///data/data/org.nativescript.ElHakawati/files/app/tns_modules/nativescript-downloader/downloader.js', line: 17, column: 42
System.err: 	Frame: function:'', file:'file:///data/data/org.nativescript.ElHakawati/files/app/app.js', line: 45, column: 38
System.err: 	Frame: function:'require', file:'', line: 1, column: 266
System.err: 	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4710)
System.err: 	at android.app.ActivityThread.-wrap1(ActivityThread.java)
System.err: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1405)
System.err: 	at android.os.Handler.dispatchMessage(Handler.java:102)
System.err: 	at android.os.Looper.loop(Looper.java:148)
System.err: 	at android.app.ActivityThread.main(ActivityThread.java:5417)
System.err: 	at java.lang.reflect.Method.invoke(Native Method)
System.err: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
System.err: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
System.err: Caused by: com.tns.NativeScriptException: 
System.err: Error calling module function 
System.err: TypeError: Cannot read property 'init' of undefined
System.err: File: "file:///data/data/org.nativescript.ElHakawati/files/app/tns_modules/nativescript-downloader/downloader.js, line: 17, column: 41
System.err: StackTrace: 
System.err: 	Frame: function:'Downloader.init', file:'file:///data/data/org.nativescript.ElHakawati/files/app/tns_modules/nativescript-downloader/downloader.js', line: 17, column: 42
System.err: 	Frame: function:'', file:'file:///data/data/org.nativescript.ElHakawati/files/app/app.js', line: 45, column: 38
System.err: 	Frame: function:'require', file:'', line: 1, column: 266
System.err: TypeError: Cannot read property 'init' of undefined
System.err: File: "<unknown>, line: 1, column: 265
System.err: StackTrace: 
System.err: 	Frame: function:'Downloader.init', file:'file:///data/data/org.nativescript.ElHakawati/files/app/tns_modules/nativescript-downloader/downloader.js', line: 17, column: 42
System.err: 	Frame: function:'', file:'file:///data/data/org.nativescript.ElHakawati/files/app/app.js', line: 45, column: 38
System.err: 	Frame: function:'require', file:'', line: 1, column: 266
System.err: 	at com.tns.Runtime.runModule(Native Method)
System.err: 	at com.tns.Runtime.runModule(Runtime.java:530)
System.err: 	at com.tns.Runtime.run(Runtime.java:522)
System.err: 	at com.tns.NativeScriptApplication.onCreate(NativeScriptApplication.java:19)
System.err: 	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1013)
System.err: 	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4707)
System.err: 	... 8 more
ActivityManager: Process org.nativescript.ElHakawati (pid 22001) has died
```

Thanks

**Comment by kefahB** [comment]
+1

**Comment by kefahB** [comment]
The problem of Downloader.init () has been fixed by following @triniwiz  post here [#2](https://github.com/triniwiz/nativescript-downloader/issues/2#issuecomment-359083967)

But the following code does not return anything !

```TS
  generateDownloads() {

    this.downloadManager = new Downloader();

    this.fileDownloaderId = this.downloadManager.createDownload({ 
      url: this.url, 
      path: this.path,
      fileName: this.destFileName 
    });

    console.log(`File Id :${this.fileDownloaderId} `);
  }
```

I put an alert () in the downloader.android.js file, it worked correctly before the 
```TS
alert();
var task = this.manager.create(request);
```
 then once after as shown by the following code the alert does not work anymore!

```TS
        var task = this.manager.create(request);
        alert(taskId)
        path = fs.path.join(request.getFilePath(), request.getFileName());
        this.taskIds.set(task, taskId);
        this.downloads.set(taskId, task);
        this.downloadRequests.set(taskId, request);
        this.downloadsData.set(taskId, {
            status: downloader_common_1.StatusCode.PENDING
        });
        return taskId;
```
